File: stats.py
class StatisticsCalculator:
    def __init__(self, data):
        self.data = data

    def mean(self):
        return sum(self.data) / len(self.data)

    def median(self):
        sorted_data = sorted(self.data)
        n = len(sorted_data)
        mid = n // 2
        if n % 2 == 0:
            return (sorted_data[mid - 1] + sorted_data[mid]) / 2
        else:
            return sorted_data[mid]

    def mode(self):
        counts = {}
        for num in self.data:
            counts[num] = counts.get(num, 0) + 1
        max_count = max(counts.values())
        return [num for num, count in counts.items() if count == max_count]

# Example usage
if __name__ == "__main__":
    data = [1, 2, 3, 3, 3, 4, 5, 6, 7, 8, 9]
    stats = StatisticsCalculator(data)
    print(f"Mean: {stats.mean():.2f}")
    print(f"Median: {stats.median():.2f}")
    print(f"Mode(s): {stats.mode()}")

File: stats.ml
let mean lst =
  let sum = List.fold_left ( + ) 0 lst in
  float_of_int sum /. float_of_int (List.length lst)

let median lst =
  let sorted = List.sort compare lst in
  let len = List.length sorted in
  if len mod 2 = 0 then
    let mid1 = List.nth sorted ((len / 2) - 1) in
    let mid2 = List.nth sorted (len / 2) in
    (float_of_int (mid1 + mid2)) /. 2.0
  else
    float_of_int (List.nth sorted (len / 2))

let mode lst =
  let sorted = List.sort compare lst in
  let counts =
    List.fold_left
      (fun acc x ->
         let count = try List.assoc x acc with Not_found -> 0 in
         (x, count + 1) :: List.remove_assoc x acc)
      [] sorted
  in
  let max_count = List.fold_left (fun acc (_, c) -> max acc c) 0 counts in
  List.map fst (List.filter (fun (_, c) -> c = max_count) counts)

(* Example usage *)
let () =
  let data = [1; 2; 3; 3; 3; 4; 5; 6; 7; 8; 9] in
  Printf.printf "Mean: %.2f\n" (mean data);
  Printf.printf "Median: %.2f\n" (median data);
  Printf.printf "Mode(s): %s\n"
    (String.concat " " (List.map string_of_int (mode data)))
